backend/src/apiserver/main.go

```diff
@@ -106,6 +106,13 @@ func main() {
 	}
 	log.SetLevel(level)
 
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer cancel()
+	err = resourceManager.SyncSwfCrs(ctx)
```

Comment:
The persistence agent seems to already have a reconcile loop for scheduled workflows. If I'm reading the code right, on start up, it'll reconcile everything and then handle creates, updates, and deletes.

Could the migration logic be added to the persistence agent instead?

Reply:
It probably could. But that doesn't sound like a persistence agent's responsibility. It doesn't sound like API server's responsibility too, but I think it fits better there since we have a [`jobStore`](https://github.com/kubeflow/pipelines/blob/953426db9895a64c380370be19c716d009a10250/backend/src/apiserver/resource/resource_manager.go#L578).

My original plan was to do it in https://github.com/kubeflow/pipelines/tree/master/backend/src/crd/controller/scheduledworkflow, but we would have to make HTTP calls to the API server. Then we decided to leave it in the API server.

It would be nice to hear others' opinions about that.

Reply:
@hbelmiro I agree that persistence agent isn't the best fit. That controller you linked could be a good fit.

I think we should consider adding a KFP version column to the `jobs` table so that you can skip generating and updating scheduled workflows if the version matches.

If were to pursue using the scheduled workflow controller, here is an idea:

* Have the scheduled workflow controller query the API server health endpoint at start up to get the `tag_name` value to see what version of KFP we are on. In the background, it could keep querying the API server to see if the version changed.
* The scheduled workflow controller's reconcile loop checks an annotation of `pipelines.kubeflow.org/version` and if the value of that annotation doesn't match `tag_name`, then the workflow definition is updated and the annotation value is set to the current version.
* When the API server creates a `ScheduledWorkflow` object, it sets the `pipelines.kubeflow.org/version` annotation.

Reply:
Sure, no problem! What do you think of the other comnment of adding a KFP version column to the jobs table so that you can skip generating and updating scheduled workflows if the version matches?

Reply:
I like that too. But other than https://github.com/kubeflow/pipelines/blob/master/VERSION, I'm not aware of a field/file from where I can get the version. Are you?
Otherwise, maybe @HumairAK and @chensun can recommend something.

Reply:
@mprahl where do you see it?